BlazarService/src/main/java/com/hubspot/blazar/util/GithubStatusHandler.java

```
@@ -0,0 +1,115 @@
+
+  private final Map<String, GitHub> gitHubByHost;
+  private final BuildDefinitionService buildDefinitionService;
+  private static final Logger LOG = org.slf4j.LoggerFactory.getLogger(GithubStatusHandler.class);
+
+  @Inject
+  public GithubStatusHandler (EventBus eventBus, Map<String, GitHub> gitHubByHost, BuildDefinitionService buildDefinitionService) {
+    this.gitHubByHost = gitHubByHost;
+    this.buildDefinitionService = buildDefinitionService;
+
+    eventBus.register(this);
+  }
+
+  @Subscribe
+  public void handleBuildStateChange(Build build) throws IOException {
+    LOG.info("Got event for a buildId: {}", build.getBuildNumber());
+    if (build.getState() == State.QUEUED) {
+      return;
+    }
+
+    BuildDefinition definition = buildDefinitionService.getByModule(build.getModuleId()).get();
+    String whitelist = Objects.firstNonNull(System.getenv("BUILD_WHITELIST"), "");
+    List<String> whitelistedRepos= Splitter.on(',').omitEmptyStrings().splitToList(whitelist);
+
+    if (!whitelistedRepos.contains(definition.getGitInfo().getRepository())) {
+      return;
+    }
+
+    String uiHost = "https://tools.hubteam.com";
+    String appRoot = "blazar";
+    String gitHost = definition.getGitInfo().getHost();
+    String gitOrg = definition.getGitInfo().getOrganization();
+    String gitRepo = definition.getGitInfo().getRepository();
+    String gitBranch  = definition.getGitInfo().getBranch();
+    String moduleName = definition.getModule().getName();
+    String buildNumber = Integer.toString(build.getBuildNumber());
+
+    String url = uiHost + "/" + appRoot + "/builds" + "/" + gitHost + "/" + gitOrg + "/" + gitRepo + "/" + gitBranch + "/" + moduleName + "/" + buildNumber;
+    GHCommitState state = toGHCommitState(build.getState());
+    String sha = build.getSha().get();
+    String description = getStateDescription(build.getState());
+    String context = "CI-blazar";
+
+    GitHub gitHubApi = gitHubByHost.get(gitHost);
```

I usually `Preconditions.checkNotNull(gitHubByHost.get(gitHost));` so it fails fast